### DISCUSSION TITLE
How to solve Next.js development build too slow problem?

### DISCUSSION THREAD

**Discussion by yaolifeng0629** [vercel/next.js#68017] [Help]
### Summary

Next Little White One:

Next 14 project development starts too slowly. How to solve it? 

unbearable, 😅

![image](https://github.com/user-attachments/assets/69aeb4da-9f26-44a9-9984-1318bf84cce4)

**Comment by icyJoseph** [comment]
🤔 that is remarkably slow, even a Next.js project started on StackBlitz, boots up somewhat faster:

<img width="829" alt="Screenshot 2024-07-22 at 10 13 54" src="https://github.com/user-attachments/assets/ab6110cf-dc4e-4a7f-bf67-58b186828fd4">

Maybe you have complex TypeScript types?

**Reply by yaolifeng0629** [reply]
@icyJoseph Nothing, they are simple types. 

This can be annoying, and the wait time for each development build can be unbearable. 😥

**Reply by icyJoseph** [reply]
Maybe some icon library? Those can be quite annoying, as they often import the entire set of icons...

If you remove most things, and just render a div, does it improve at all? What about a newly made project?

**Reply by yaolifeng0629** [reply]
Yes, You're right, But now I have introduced it on demand.

The development environment is still building slowly.
```sh
"tailwindcss-animated": "^1.0.1",
"@iconify-json/fa6-brands": "^1.1.18",
"@iconify-json/logos": "^1.1.42",
"@iconify-json/skill-icons": "^1.1.8",
"@radix-ui/react-tooltip": "^1.0.7",
"@radix-ui/react-toast": "^1.1.5",
"@radix-ui/react-slot": "^1.0.2",
...
```

**Reply by icyJoseph** [reply]
Could you try to use this:

- https://nextjs.org/docs/app/api-reference/next-config-js/optimizePackageImports

There was also another way, but I can't find the docs page.

What do you mean, "on demand"?

**Reply by yaolifeng0629** [reply]
Import only use to package, not all

**Reply by yaolifeng0629** [reply]
I just found the questions that feed back on this issues. #48748
